hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/net/TestNetUtils.java

```diff
@@ -801,6 +802,48 @@ public void testBindToLocalAddress() throws Exception {
         .bindToLocalAddress(NetUtils.getLocalInetAddress("127.0.0.1"), true));
   }
 
+  public static class WrappedIOException extends IOException {
+    public WrappedIOException(String msg, Throwable cause) {
+      super(msg, cause);
+    }
+  }
+
+  private static class PrivateIOException extends IOException {
+    PrivateIOException(String msg, Throwable cause) {
+      super(msg, cause);
+    }
+  }
+
+  @Test
+  public void testAddNodeNameToIOException() {
+    IOException e0 = new IOException("test123");
+    assertNull(e0.getCause());
```

Comment:
can we have assertj here..as we do that move to junit5, best to start off with the cross-version test asserts.

actually, given there's so many assertNull(throwable.getCause(), what about a new method `assertNullCause(Throwable)` and use through this